jobs/kpi-forecasting/kpi-forecasting/pandas_extras.py

```diff
@@ -0,0 +1,29 @@
+import numpy as np
+import pandas as pd
+
+
+def percentile(p: int = 50, name_format: str = "p{:02.0f}"):
+    """A method to calculate percentiles along dataframe axes via the `pandas.agg` method."""
+
+    def f(x):
+        return x.quantile(p / 100)
+
+    f.__name__ = name_format.format(p)
+    return f
+
+
+def aggregate_to_period(
+    df: pd.DataFrame,
+    period: str,
+    aggregation: callable = np.sum,
+    date_col: str = "submission_date",
+) -> pd.DataFrame:
+    """Floor dates to the correct period and aggregate."""
+    if period.lower() in ["day", "month", "year"]:
```

Comment:
Could avoid the if, else if you flipped around the condition `not in` and if True raise the exception. and if we make it through the block do your assignment. Looks like it should always happen.

Reply:
💯 clever! done.